packages/block-editor/src/components/colors-gradients/control.js

```diff
@@ -78,22 +77,24 @@ function ColorGradientControlInner( {
 						</legend>
 					) }
 					{ canChooseAColor && canChooseAGradient && (
-						<ToggleGroupControl
-							value={ currentTab }
-							onChange={ setCurrentTab }
-							label={ __( 'Select color type' ) }
-							hideLabelFromVision
-							isBlock
+						<TabPanel
+							className="block-editor-color-gradient-control__tabs"
+							onSelect={ setCurrentTab }
+							tabs={ [
+								{
+									name: 'color',
+									title: 'Solid color',
+									value: 'color',
+								},
+								{
+									name: 'gradient',
+									title: 'Gradient',
+									value: 'gradient',
+								},
+							] }
 						>
-							<ToggleGroupControlOption
-								value="color"
-								label={ __( 'Solid' ) }
-							/>
-							<ToggleGroupControlOption
-								value="gradient"
-								label={ __( 'Gradient' ) }
-							/>
-						</ToggleGroupControl>
+						{ ( tab ) => <p className="screen-reader-text">Selected tab: { tab.title }</p> }
```

Comment:
This may be a bit of an early optimization in the PR, but we could extract the values of the `tabs` to a separate variables, in order to avoid re-creating the same object every render (and to re-use the variable across the component instead of hardcoding the tab IDs around the code).

<details>

<summary>For example</summary>

```diff
diff --git a/packages/block-editor/src/components/colors-gradients/control.js b/packages/block-editor/src/components/colors-gradients/control.js
index 8d38e8e901..f58be3dc82 100644
--- a/packages/block-editor/src/components/colors-gradients/control.js
+++ b/packages/block-editor/src/components/colors-gradients/control.js
@@ -29,6 +29,19 @@ const colorsAndGradientKeys = [
 	'disableCustomGradients',
 ];
 
+const TAB_COLOR = {
+	name: 'color',
+	title: 'Solid color',
+	value: 'color',
+};
+const TAB_GRADIENT = {
+	name: 'gradient',
+	title: 'Gradient',
+	value: 'gradient',
+};
+
+const TABS_SETTINGS = [ TAB_COLOR, TAB_GRADIENT ];
+
 function ColorGradientControlInner( {
 	colors,
 	gradients,
@@ -52,7 +65,9 @@ function ColorGradientControlInner( {
 		onGradientChange &&
 		( ! isEmpty( gradients ) || ! disableCustomGradients );
 	const [ currentTab, setCurrentTab ] = useState(
-		gradientValue ? 'gradient' : !! canChooseAColor && 'color'
+		gradientValue
+			? TAB_GRADIENT.value
+			: !! canChooseAColor && TAB_COLOR.value
 	);
 
 	if ( ! canChooseAColor && ! canChooseAGradient ) {
@@ -80,23 +95,13 @@ function ColorGradientControlInner( {
 						<TabPanel
 							className="block-editor-color-gradient-control__tabs"
 							onSelect={ setCurrentTab }
-							tabs={ [
-								{
-									name: 'color',
-									title: 'Solid color',
-									value: 'color',
-								},
-								{
-									name: 'gradient',
-									title: 'Gradient',
-									value: 'gradient',
-								},
-							] }
+							tabs={ TABS_SETTINGS }
 						>
 						{ ( tab ) => <p className="screen-reader-text">Selected tab: { tab.title }</p> }
 						</TabPanel>
 					) }
-					{ ( currentTab === 'color' || ! canChooseAGradient ) && (
+					{ ( currentTab === TAB_COLOR.value ||
+						! canChooseAGradient ) && (
 						<ColorPalette
 							value={ colorValue }
 							onChange={
@@ -118,7 +123,8 @@ function ColorGradientControlInner( {
 							enableAlpha={ enableAlpha }
 						/>
 					) }
-					{ ( currentTab === 'gradient' || ! canChooseAColor ) && (
+					{ ( currentTab === TAB_GRADIENT.value ||
+						! canChooseAColor ) && (
 						<GradientPicker
 							value={ gradientValue }
 							onChange={
```
</details>

Reply:
I'll make these changes 👍